### PR TITLE
Standardize group arguments in functions

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR aims to tackle part of https://github.com/tidymodels/probably/issues/104. Change 1 specifically.

While it might seems strange, I'm sticking with `group` over `.by`, as it is consistent with the existing codebase. Once this is done, I will swap everything over to `.by` as part of Change 2.

- [x] add `group` argument to `cal_estimate_*.data.frame()` functions
- [x] add `group` argument to `cal_plot_.*.data.frame()` functions
- [x] Handle `.config` in call_estimate_*.tune_results()` functions
- [x] Handle `.config` in call_plot_*.tune_results()` functions

### CODE REVIEW

File: R/cal-estimate-beta.R [hunk 1]
```diff
@@ -34,8 +34,13 @@ cal_estimate_beta.data.frame <- function(.data,
...
                                          estimate = dplyr::starts_with(".pred_"),
                                          parameters = NULL,
-                                         ...) {
+                                         ...,
+                                         group = NULL) {
   stop_null_parameters(parameters)
+
+  check_group_argument({{ group }}, .data)
+  .data <- dplyr::group_by(.data, dplyr::across({{ group }}))
+
```
Comment (EmilHvitfeldt) [inline R/cal-estimate-beta.R:43]: All `*.data.frame()` functions calls `check_group_argument()` and applies `dplyr::group_by()` with the group. This does nothing if `group = NULL`

File: R/cal-plot-breaks.R [hunk 2]
```diff
@@ -134,23 +133,26 @@ cal_plot_breaks.data.frame <- function(.data,
...
-    group = {{ group }},
+    event_level = event_level,
+    ...
+  )
+
+  cal_plot_breaks_impl(
+    .data = tune_args$predictions,
+    truth = !!tune_args$truth,
+    estimate = !!tune_args$estimate,
+    group = !!tune_args$group,
```
Comment (EmilHvitfeldt) [inline R/cal-plot-breaks.R:155]: All `*tune.results()` are changed to use `tune_results_args()`.

File: R/cal-plot-regression.R [hunk 3]
```diff
@@ -95,6 +91,10 @@ regression_plot_impl <- function(.data, truth, estimate, group,
   estimate <- enquo(estimate)
   group <- enquo(group)
 
+  if (quo_is_null(group)) {
+    .data[[".config"]] <- NULL
+  }
```
Comment (EmilHvitfeldt) [inline R/cal-plot-regression.R:96]: If there isn't a group variable, e.i. `.config` is constant, it is removed from the plotting data.

This makes it easier to test that faceting isn't done with constant `.config`

File: R/cal-plot-utils.R [hunk 4]
```diff
@@ -250,15 +248,17 @@ tune_results_args <- function(.data,
     estimate <- expr(dplyr::starts_with(".pred"))
   }
 
-  if (quo_is_null(group)) {
+  if (dplyr::n_distinct(.data[[".predictions"]][[1]][[".config"]]) > 1) {
     group <- quo(.config)
+  } else {
+    group <- quo(NULL)
   }
```
Comment (EmilHvitfeldt) [inline R/cal-plot-utils.R:255]: this is the part that detects if there are more than 1 unique `.config`

**Comment by topepo** [comment]
For multinomial outcomes, much (but not all) of the infrastructure make a series of 1:all problems and applies binomial methods. 

While the automatic detection of `.config` works for binomial, it doesn't find it for multinomial: 

``` r
hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"



library(tidymodels)
library(probably)
#> 
#> Attaching package: 'probably'
#> The following objects are masked from 'package:base':
#> 
#>     as.factor, as.ordered
```

``` r

tidymodels_prefer()
theme_set(theme_bw())
options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
```

``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
```



``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
```



<sup>Created on 2023-05-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

(using `group = .config` does work though).

**Comment by topepo** [comment]
For multinomial outcomes, much (but not all) of the infrastructure make a series of 1:all problems and applies binomial methods. 

While the automatic detection of `.config` works for binomial, it doesn't find it for multinomial: 

``` r
hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"



library(tidymodels)
library(probably)
#> 
#> Attaching package: 'probably'
#> The following objects are masked from 'package:base':
#> 
#>     as.factor, as.ordered
```

``` r

tidymodels_prefer()
theme_set(theme_bw())
options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
```

``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
```



``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
```



<sup>Created on 2023-05-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

(using `group = .config` does work though).

**Comment by topepo** [comment]
For multinomial outcomes, much (but not all) of the infrastructure make a series of 1:all problems and applies binomial methods. 

While the automatic detection of `.config` works for binomial, it doesn't find it for multinomial: 

``` r
hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"



library(tidymodels)
library(probably)
#> 
#> Attaching package: 'probably'
#> The following objects are masked from 'package:base':
#> 
#>     as.factor, as.ordered
```

``` r

tidymodels_prefer()
theme_set(theme_bw())
options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
```

``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
```



``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
```



<sup>Created on 2023-05-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

(using `group = .config` does work though).

**Comment by topepo** [comment]
For multinomial outcomes, much (but not all) of the infrastructure make a series of 1:all problems and applies binomial methods. 

While the automatic detection of `.config` works for binomial, it doesn't find it for multinomial: 

``` r
hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
#> Error in hpc_cv %>% mutate(.config = sample(letters[1:2], nrow(hpc_cv), : could not find function "%>%"



library(tidymodels)
library(probably)
#> 
#> Attaching package: 'probably'
#> The following objects are masked from 'package:base':
#> 
#>     as.factor, as.ordered
```

``` r

tidymodels_prefer()
theme_set(theme_bw())
options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
```

``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
```



``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
```



<sup>Created on 2023-05-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

(using `group = .config` does work though).

**Comment by EmilHvitfeldt** [comment]
That should be fixed now 

``` r
library(tidymodels)
library(probably)
#> 
#> Attaching package: 'probably'
#> The following objects are masked from 'package:base':
#> 
#>     as.factor, as.ordered

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_windowed(truth = obs, VF:L)
```



``` r

hpc_cv %>%
  mutate(.config = sample(letters[1:2], nrow(hpc_cv), replace = TRUE)) %>% 
  cal_plot_breaks(truth = obs, VF:L)
```



<sup>Created on 2023-05-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Pull request merged**